src/langchain_google_spanner/loader.py

```diff
@@ -0,0 +1,407 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import json
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Union
+
+from google.cloud.spanner import Client, KeySet  # type: ignore
+from google.cloud.spanner_admin_database_v1.types import DatabaseDialect  # type: ignore
+from google.cloud.spanner_v1.data_types import JsonObject  # type: ignore
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_core.documents import Document
+
```

Comment:
Add user agent
```suggestion
from .version import __version__

USER_AGENT_LOADER = "langchain-google-spanner-python:document_loader" + __version__
```
Then update the logic for the client to append the user agent
https://github.com/googleapis/langchain-google-datastore-python/blob/3a62380794ad567ff8e9c0d5d7054093fb54cdb4/src/langchain_google_datastore/document_loader.py#L160-L168